src/frontend/src/components/authenticateBox.ts

```diff
@@ -45,52 +45,48 @@ type ChasmOpts = {
 
 /** Authentication box component which authenticates a user
  * to II or to another dapp */
-export const authenticateBox = (
+export const authenticateBox = async (
   connection: Connection,
   templates: AuthTemplates
 ): Promise<LoginData> => {
-  const retryOnError = async (result: LoginFlowResult): Promise<LoginData> => {
+  const promptAuth = () =>
+    new Promise<LoginFlowResult>((resolve) => {
+      authenticateBoxTemplate({
+        templates,
+        addDevice: () => addRemoteDevice(connection),
+        onContinue: (userNumber) => {
+          resolve(authenticate(connection, userNumber));
+        },
+        register: () => {
+          resolve(registerIfAllowed(connection));
+        },
+        recoverAnchor: (userNumber) => useRecovery(connection, userNumber),
+        userNumber: getUserNumber(),
+      });
+    });
+
+  // Retry until user has successfully authenticated
+  for (;;) {
```

Comment:
equivalent to `while(true)` but has the advantage of being accepted by eslint by default: https://github.com/eslint/eslint/issues/5477